aten/src/ATen/native/nested/NestedTensorMath.cpp

```diff
@@ -557,5 +557,39 @@ Tensor& NestedTensor_mul__Tensor(Tensor& self, const Tensor& other) {
       });
 }
 
+Tensor select_nested(const Tensor& self, int64_t dim, int64_t index) {
+  TORCH_CHECK(
+    dim == 0,
+    "NestedTensor can only be selected along dimension 0 ",
+    "got dimension ", dim, " instead."
+  );
+  // size_matrix.dim() can only have 2 possible values:
+  //     * 2, as expected
+  //     * 0, when the nested tensor is empty
+  Tensor sizemat = get_nested_size_tensor(self);
+  TORCH_CHECK(
+    sizemat.dim() == 2,
+    "cannot index an empty nested tensor."
+  );
+  // nested_tensor[i] = i-th original tensor
+  int64_t ntensors = sizemat.size(0);
+  TORCH_CHECK_INDEX(
+    index >= -ntensors && index < ntensors,
+    "index ", index, " is out of bounds for nested tensor dimension 0 with size ", ntensors);
+  int64_t positive_index = index < 0 ? index + ntensors : index;
+  // buffer contains the underlying data in a contiguous vector
+  Tensor buffer = get_buffer(self);
+  // determine the memory section of the i-th original tensor
```

Comment:
Nit: The calculation you have in mind here, converting from a nested size to a list of offsets, conceptually already exists in [NestedTensorTransformerFunctions.cpp](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/nested/NestedTensorTransformerFunctions.cpp#L88-L105). Without wanting to push for premature optimization (i.e. no need to address this) note that it is much more efficient to operate on the nested size directly due to overhead from calling `at::prod` and the fact that sizemat is rather small.

Reply:
It'd also be a useful utility for unbind [above in this file](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/nested/NestedTensorMath.cpp#L113-L143).